jetty-core/jetty-quic/jetty-quic-server/src/main/java/org/eclipse/jetty/quic/server/ServerQuicSession.java

```diff
@@ -44,12 +45,14 @@
 public class ServerQuicSession extends QuicSession implements CyclicTimeouts.Expirable
 {
     private final Connector connector;
+    private final ServerQuicConfiguration quicConfiguration;
     private long expireNanoTime = Long.MAX_VALUE;
 
-    public ServerQuicSession(Executor executor, Scheduler scheduler, ByteBufferPool bufferPool, QuicheConnection quicheConnection, QuicConnection connection, SocketAddress remoteAddress, Connector connector)
+    public ServerQuicSession(Executor executor, Scheduler scheduler, ByteBufferPool bufferPool, QuicheConnection quicheConnection, QuicConnection connection, SocketAddress remoteAddress, Connector connector, ServerQuicConfiguration quicConfiguration)
```

Comment:
You can change this constructor to take a `ServerQuicConnection` instead of a `QuicConnection`, so that you can access the `ServerQuicConnection` to get the configuration.

Reply:
Let me change `ServerQuicSession`'s constructor to take a `ServerQuicConnection`; that will make sure the already existing `getQuicConnection()` method that returns a `ServerQuicConnection` never throws a `ClassCastException`.